ersilia/core/tracking.py

```diff
@@ -225,166 +242,170 @@ def upload_to_cddvault(output_df, api_key):
+        self.lake_folder = os.path.join(self.ersilia_runs_folder, "lake")
+        os.makedirs(self.lake_folder, exist_ok=True)
+        self.logs_folder = os.path.join(self.ersilia_runs_folder, "logs")
+        os.makedirs(self.logs_folder, exist_ok=True)
+        self.tabular_result_logger = TabularResultLogger()
+        self.PERSISTENT_FILE_PATH = open_persistent_file(model_id)
+        
 
-    # function to be called before model is run
     def start_tracking(self):
         """
-        Runs any code necessary for the beginning of the run.
-        Currently necessary for tracking the runtime and memory usage of a run.
-        """
+	Runs any code necessary for the beginning of the run.
+	Currently necessary for tracking the runtime and memory usage of a run.
+	
+	"""
         self.time_start = datetime.now()
         tracemalloc.start()
         self.memory_usage_start = tracemalloc.get_traced_memory()[0]
-
-    def sample_df(self, df, num_rows, num_cols):
-        """
-        Returns a sample of the dataframe, with the specified number of rows and columns.
+    
+    def calculate_stats(self, result):
         """
-        return df.sample(num_rows, axis=0).sample(num_cols, axis=1)
-
-    def stats(self, result):
-        """
-        Stats function: calculates the basic statistics of the output file from a model. This includes the
+        calculate_stats function: calculates the basic statistics of the output file from a model. This includes the
         mode (if applicable), minimum, maximum, and standard deviation.
 
         :param result: The path to the model's output file.
         :return: A dictionary containing the stats for each column of the result.
         """
 
-        dat = pd.read_csv(result)
+        df = pd.read_csv(result)
 
         # drop first two columns (key, input)
-        dat = dat.drop(["key", "input"], axis=1)
+        df = df.drop(["key", "input"], axis=1)
 
-        # calculate statistics
+        # calculate statistics	
         stats = {}
-        for column in dat:
-            column_stats = {}
-            column_stats["mean"] = dat[column].mean()
-            if len(dat[column].mode()) == 1:
-                column_stats["mode"] = dat[column].mode().iloc[0]
-            else:
-                column_stats["mode"] = None
-            column_stats["min"] = dat[column].min()
-            column_stats["max"] = dat[column].max()
-            column_stats["std"] = dat[column].std()
-
-            stats[column] = column_stats
-
-        return stats
-
-    def get_file_sizes(self, input_df, output_df):
+        for column in df:
+            stats[column] = {
+                "mean": df[column].mean(),
+                "mode": df[column].mode().iloc[0] if not df[column].mode().empty else None,
+                "min": df[column].min(),
+                "max": df[column].max(),
+                "std": df[column].std(),
+            }
+        return stats    
+
+    def get_file_sizes(self, df):
```

Comment:
Same as above - don't modify this method in this PR. Unrelated. Being worked on by @dzumii